components/XElement/XElement.astro

```diff
@@ -137,13 +141,16 @@ for (const name in attrs) {
 
 const onLoadString: string = (
     listenerString || listenerOfOnce
-        ? `(async function($$,$){$=this;$$=(await import($$.src)).default;${
+        ? (await transformWithEsbuild(`(async function($$,$){$=this;$$=(await import($$.src)).default;${
             propsString ?? ``
         }${
             listenerString ?? ``
         }${
             listenerOfOnce ?? ``
-        }}).call(this.previousSibling,this.remove()||this)`
+        }}).call(this.previousSibling,this.remove()||this)`,`${tag}_Onload.ts`,{
+			loader:'ts',
+			minify:true
```

Comment:
See I am debating where to have this switched on only in `import.meta.env.PROD` and not in `DEV`, This is open to discussion,

Reply:
You don't need loader "ts", it will automatically pickup that the file is typescript from the file extension

Reply:
ah awesome ill remove that out, 
what is your thoughts on only minifying on PROD and not DEV?

Reply:
Not everyone sets environmental variables when pushing to production, so, you may just want minimization on for everything

Reply:
I was considering that might be the counter arguement, thanks for reaffriming that for me, I will do, =)